File: tools/pas2php-code-6/pas2php_include/system.php
<?PHP
// 
// Converted From Pascal By Pas2Php v0.8
// Converts Object Pascal to PHP
// http://www.wascal.net/?page=pas2php
// Pas2Php v0.8 is Copyright (C) 2014-2016 Derek John Evans
// 
include_once('system.php');
include_once('p2pvariants.php');
include_once('phplib.php');
include_once('sysutils.php');
include_once('variants.php');
// MaxInt = $7fffffff;
define("MaxInt", 2147483647);
// DirectorySeparator = DIRECTORY_SEPARATOR;
define("DirectorySeparator", DIRECTORY_SEPARATOR);
// ExtensionSeparator = '.';
define("ExtensionSeparator", ".");
// DriveSeparator = ':';
define("DriveSeparator", ":");
// PathSeparator = ';';
define("PathSeparator", ";");
// LineEnding = #13#10;
define("LineEnding", "\r\n");
// MaxPathLen = 260;
define("MaxPathLen", 260);
// AllFilesMask = '*';
define("AllFilesMask", "*");
// Null = null;
define("Null", null);
// Unassigned = null;
define("Unassigned", null);
class TObject {
  function __get($name) {
    return parent::__get($name);
  }
  function __set($name, $value) {
    return parent::__set($name, $value);
  }
  // constructor Create;
  function __construct() {
  }
  // destructor Destroy;  Virtual;
  function __destruct() {
  }
  // procedure Free;
  function Free() {
  }
  // function ClassName : string;
  function ClassName() {
    // Result : string;
    $Result = "";
    $Result = get_class($this);
    return $Result;
  }
}
// function string(const AVariant: variant) : string;
function string($AVariant) {
  // Result : string;
  $Result = "";
  $Result = strval($AVariant);
  return $Result;
}
// function shortstring(const AVariant: variant) : string;
function shortstring($AVariant) {
  // Result : string;
  $Result = "";
  $Result = strval($AVariant);
  return $Result;
}
// function ansistring(const AVariant: variant) : string;
function ansistring($AVariant) {
  // Result : string;
  $Result = "";
  $Result = strval($AVariant);
  return $Result;
}
// function widestring(const AVariant: variant) : string;
function widestring($AVariant) {
  // Result : string;
  $Result = "";
  $Result = strval($AVariant);
  return $Result;
}
// function char(const AInteger: Integer) : string;
function char($AInteger) {
  // Result : string;
  $Result = "";
  $Result = chr($AInteger);
  return $Result;
}
// function integer(const AVariant: variant) : Integer;
function integer($AVariant) {
  // Result : Integer;
  $Result = 0;
  $Result = intval($AVariant, 10);
  return $Result;
}
// function int64(const AVariant: variant) : Integer;
function int64($AVariant) {
  // Result : Integer;
  $Result = 0;
  $Result = intval($AVariant, 10);
  return $Result;
}
// function smallint(const AVariant: variant) : Integer;
function smallint($AVariant) {
  // Result : Integer;
  $Result = 0;
  $Result = intval($AVariant, 10);
  return $Result;
}
// function longint(const AVariant: variant) : Integer;
function longint($AVariant) {
  // Result : Integer;
  $Result = 0;
  $Result = intval($AVariant, 10);
  return $Result;
}
// function single(const AVariant: variant) : extended;
function single($AVariant) {
  // Result : extended;
  $Result = 0.0;
  $Result = floatval($AVariant);
  return $Result;
}
// function double(const AVariant: variant) : extended;
function double($AVariant) {
  // Result : extended;
  $Result = 0.0;
  $Result = floatval($AVariant);
  return $Result;
}
// function extended(const AVariant: variant) : extended;
function extended($AVariant) {
  // Result : extended;
  $Result = 0.0;
  $Result = floatval($AVariant);
  return $Result;
}
// function real(const AVariant: variant) : extended;
function real($AVariant) {
  // Result : extended;
  $Result = 0.0;
  $Result = floatval($AVariant);
  return $Result;
}
// function variant(const AVariant: variant) : variant;
function variant($AVariant) {
  // Result : variant;
  $Result = null;
  $Result = $AVariant;
  return $Result;
}
// function Assigned(const AVariant: variant) : Boolean;
function Assigned($AVariant) {
  // Result : Boolean;
  $Result = false;
  $Result = !is_null($AVariant);
  return $Result;
}
// procedure Enforce(const ATrue: Boolean;const AMessage: string);
function Enforce($ATrue, $AMessage = "Enforce") {
  if (!$ATrue) {
    throw new EAssertionFailed($AMessage);
  }
}
// function Trunc(const AExtended: extended) : extended;
function Trunc($AExtended) {
  // Result : extended;
  $Result = 0.0;
  $Result = intval($AExtended, 10);
  return $Result;
}
// function SystemOrd(const AVariant: variant) : Integer;
function SystemOrd($AVariant) {
  // Result : Integer;
  $Result = 0;
  if (is_string($AVariant)) {
    $Result = ord($AVariant);
  }
  else {
    if (is_int($AVariant)) {
      $Result = $AVariant;
    }
    else {
      throw new EArgumentException("Invalid Ordinal Type");
    }
  }
  return $Result;
}
// function Pred(const AVariant: variant) : variant;
function Pred($AVariant) {
  // LString : string;
  $LString = "";
  // Result : variant;
  $Result = null;
  if (is_string($AVariant)) {
    $Result = chr(ord($AVariant[0]) - 1);
  }
  else {
    $Result = $AVariant - 1;
  }
  return $Result;
}
// function Succ(const AVariant: variant) : variant;
function Succ($AVariant) {
  // Result : variant;
  $Result = null;
  if (is_string($AVariant)) {
    $Result = chr(ord($AVariant[0]) + 1);
  }
  else {
    $Result = $AVariant + 1;
  }
  return $Result;
}
// procedure Inc(var AInteger: Integer;const AOffset: Integer);
function Inc(&$AInteger, $AOffset = 1) {
  $AInteger = $AInteger + $AOffset;
}
// procedure Dec(var AInteger: Integer;const AOffset: Integer);
function Dec(&$AInteger, $AOffset = 1) {
  $AInteger = $AInteger - $AOffset;
}
// function AllowDirectorySeparators : variant;
function AllowDirectorySeparators() {
  // Result : variant;
  $Result = null;
  $Result = VarArrayOf(array("\\", "/"));
  return $Result;
}
/*
function MidStr(const AString: string;const AOffset: Integer;
               const ALength: Integer) : string;
*/
function MidStr($AString, $AOffset, $ALength) {
  // Result : string;
  $Result = "";
  $Result = substr($AString, $AOffset - 1, $ALength);
  return $Result;
}
// function Length(const AVariant: variant) : Integer;
function Length($AVariant) {
  // Result : Integer;
  $Result = 0;
  if (is_string($AVariant)) {
    $Result = strlen($AVariant);
  }
  else {
    if (is_array($AVariant)) {
      $Result = count($AVariant);
    }
    else {
      $Result = 0;
    }
  }
  return $Result;
}
/*
procedure SetLength(var AVariant: variant;const AD1: Integer;
                   const AD2: Integer;const AD3: Integer);
*/
function SetLength(&$AVariant, $AD1, $AD2 =  - 1, $AD3 =  - 1) {
  if (is_string($AVariant)) {
    if ($AD1 < strlen($AVariant)) {
      $AVariant = substr($AVariant, 0, $AD1);
    }
    else {
      $AVariant = str_pad($AVariant, $AD1);
    }
  }
  else {
    if (is_array($AVariant)) {
      VarArraySetLength($AVariant, $AD1, $AD2, $AD3);
    }
  }
}
// function Low(const AVariant: variant) : Integer;
function Low($AVariant) {
  // Result : Integer;
  $Result = 0;
  if (is_array($AVariant)) {
    $Result = 0;
  }
  else {
    if (is_string($AVariant)) {
      $Result = 1;
    }
  }
  return $Result;
}
// function High(const AVariant: variant) : Integer;
function High($AVariant) {
  // Result : Integer;
  $Result = 0;
  $Result = Length($AVariant);
  if (is_array($AVariant)) {
    $Result = $Result - 1;
  }
  return $Result;
}
// function upcase(const AString: string) : string;
function upcase($AString) {
  // Result : string;
  $Result = "";
  $Result = strtoupper($AString);
  return $Result;
}
// function lowercase(const AString: string) : string;
function lowercase($AString) {
  // Result : string;
  $Result = "";
  $Result = strtolower($AString);
  return $Result;
}
// procedure Write;
function Write() {
  // LIndex : Integer;
  $LIndex = 0;
  // AArgs : variant;
  $AArgs = null;
  $AArgs = func_get_args();
  for ($LIndex = 0, $LIndex_High_ = count($AArgs) - 1; $LIndex <= $LIndex_High_; $LIndex++) {
    echo($AArgs[$LIndex]);
  }
}
// procedure WriteLn;
function WriteLn() {
  // LIndex : Integer;
  $LIndex = 0;
  // LArgs : variant;
  $LArgs = null;
  $LArgs = func_get_args();
  for ($LIndex = 0, $LIndex_High_ = count($LArgs) - 1; $LIndex <= $LIndex_High_; $LIndex++) {
    echo($LArgs[$LIndex]);
  }
  echo(LineEnding);
}
?>

File: tools/pas2php-code-6/pas2php_include/__system.pas
{
 ***************************************************************************
 *                                                                         *
 *   This source is free software; you can redistribute it and/or modify   *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This code is distributed in the hope that it will be useful, but      *
 *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU     *
 *   General Public License for more details.                              *
 *                                                                         *
 *   A copy of the GNU General Public License is available on the World    *
 *   Wide Web at <http://www.gnu.org/copyleft/gpl.html>. You can also      *
 *   obtain it by writing to the Free Software Foundation,                 *
 *   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.        *
 *                                                                         *
 ***************************************************************************
}

unit __System;

interface

uses P2PVariants, PhpLib, SysUtils, Variants;

{$IFDEF PAS2PHP}

const
  MaxInt = $7fffffff;
  DirectorySeparator = DIRECTORY_SEPARATOR;
  ExtensionSeparator = '.';
  DriveSeparator = ':';
  PathSeparator = ';';
  LineEnding = #13#10;
  MaxPathLen = 260;
  AllFilesMask = '*';

var GLOBALS: Variant; external;
var _SERVER: Variant; external;
var _GET: Variant; external;
var _POST: Variant; external;
var _COOKIE: Variant; external;
var _FILE: Variant; external;
var _ENV: Variant; external;
var _REQUEST: Variant; external;
var _SESSION: Variant; external;

const
  Null = null;
  Unassigned = null;

type

  TObject = class
    constructor Create;
    destructor Destroy; virtual;
    procedure Free;
    function ClassName: string;
  end;

procedure return(const AVariant: variant);
function string(const AVariant: variant): string;
function shortstring(const AVariant: variant): string;
function ansistring(const AVariant: variant): string;
function widestring(const AVariant: variant): string;

{$ENDIF}

function char(const AInteger: integer): string;
function integer(const AVariant: variant): integer;
function int64(const AVariant: variant): integer;
function smallint(const AVariant: variant): integer;
function longint(const AVariant: variant): integer;
function single(const AVariant: variant): extended;
function double(const AVariant: variant): extended;
function extended(const AVariant: variant): extended;
function real(const AVariant: variant): extended;
function variant(const AVariant: variant): variant;

function AllowDirectorySeparators: variant;
function Assigned(const AVariant: variant): boolean;
function High(const AVariant: variant): integer;
function Length(const AVariant: variant): integer;
function Low(const AVariant: variant): integer;
function MidStr(const AString: string; const AOffset, ALength: integer): string;
function Pred(const AVariant: variant): variant;
function Succ(const AVariant: variant): variant;
function SystemOrd(const AVariant: variant): integer;
function Trunc(const AExtended: extended): extended;
procedure Dec(var AInteger: integer; const AOffset: integer = 1);
procedure Enforce(const ATrue: boolean; const AMessage: string = 'Enforce');
procedure Inc(var AInteger: integer; const AOffset: integer = 1);
procedure SetLength(var AVariant: variant; const AD1: integer; const AD2: integer = -1;
  const AD3: integer = -1);
procedure Write;
procedure WriteLn;
function upcase(const AString: string): string;
function lowercase(const AString: string): string;

implementation

{$IFDEF PAS2PHP}

function string(const AVariant: variant): string;
begin
  Result := php_strval(AVariant);
end;

function shortstring(const AVariant: variant): string;
begin
  Result := php_strval(AVariant);
end;

function ansistring(const AVariant: variant): string;
begin
  Result := php_strval(AVariant);
end;

function widestring(const AVariant: variant): string;
begin
  Result := php_strval(AVariant);
end;

constructor TObject.Create;
begin
end;

destructor TObject.Destroy;
begin
end;

function TObject.ClassName: string;
begin
  Result := php_get_class(Self);
end;

procedure TObject.Free;
begin
end;

{$ENDIF}

function char(const AInteger: integer): string;
begin
  Result := php_chr(AInteger);
end;

function integer(const AVariant: variant): integer;
begin
  Result := php_intval(AVariant, 10);
end;

function int64(const AVariant: variant): integer;
begin
  Result := php_intval(AVariant, 10);
end;

function smallint(const AVariant: variant): integer;
begin
  Result := php_intval(AVariant, 10);
end;

function longint(const AVariant: variant): integer;
begin
  Result := php_intval(AVariant, 10);
end;

function single(const AVariant: variant): extended;
begin
  Result := php_floatval(AVariant);
end;

function double(const AVariant: variant): extended;
begin
  Result := php_floatval(AVariant);
end;

function extended(const AVariant: variant): extended;
begin
  Result := php_floatval(AVariant);
end;

function real(const AVariant: variant): extended;
begin
  Result := php_floatval(AVariant);
end;

function variant(const AVariant: variant): variant;
begin
  Result := AVariant;
end;

function Assigned(const AVariant: variant): boolean;
begin
  Result := not php_is_null(AVariant);
end;

procedure Enforce(const ATrue: boolean; const AMessage: string);
begin
  if not ATrue then begin
    raise EAssertionFailed.Create(AMessage);
  end;
end;

function Trunc(const AExtended: extended): extended;
begin
  Result := php_intval(AExtended, 10);
end;

function SystemOrd(const AVariant: variant): integer;
begin
  if php_is_string(AVariant) then begin
    Result := php_ord(AVariant);
  end else if php_is_int(AVariant) then begin
    Result := AVariant;
  end else begin
    raise EArgumentException.Create('Invalid Ordinal Type');
  end;
end;

function Pred(const AVariant: variant): variant;
var
  LString: string;
begin
  if php_is_string(AVariant) then begin
    Result := php_chr(php_ord(AVariant[0]) - 1);
  end else begin
    Result := AVariant - 1;
  end;
end;

function Succ(const AVariant: variant): variant;
begin
  if php_is_string(AVariant) then begin
    Result := php_chr(php_ord(AVariant[0]) + 1);
  end else begin
    Result := AVariant + 1;
  end;
end;

procedure Inc(var AInteger: integer; const AOffset: integer);
begin
  AInteger := AInteger + AOffset;
end;

procedure Dec(var AInteger: integer; const AOffset: integer);
begin
  AInteger := AInteger - AOffset;
end;

function AllowDirectorySeparators: variant;
begin
  Result := VarArrayOf(['\', '/']);
end;

function MidStr(const AString: string; const AOffset, ALength: integer): string;
begin
  Result := php_substr(AString, AOffset - 1, ALength);
end;

function Length(const AVariant: variant): integer;
begin
  if php_is_string(AVariant) then begin
    Result := php_strlen(AVariant);
  end else if php_is_array(AVariant) then begin
    Result := php_count(AVariant);
  end else begin
    Result := 0;
  end;
end;

procedure SetLength(var AVariant: variant; const AD1, AD2, AD3: integer);
begin
  if php_is_string(AVariant) then begin
    if AD1 < php_strlen(AVariant) then begin
      AVariant := php_substr(AVariant, 0, AD1);
    end else begin
      AVariant := php_str_pad(AVariant, AD1);
    end;
  end else if php_is_array(AVariant) then begin
    VarArraySetLength(AVariant, AD1, AD2, AD3);
  end;
end;

function Low(const AVariant: variant): integer;
begin
  if php_is_array(AVariant) then begin
    Result := 0;
  end else if php_is_string(AVariant) then begin
    Result := 1;
  end;
end;

function High(const AVariant: variant): integer;
begin
  Result := Length(AVariant);
  if php_is_array(AVariant) then begin
    Result := Result - 1;
  end;
end;

function upcase(const AString: string): string;
begin
  Result := php_strtoupper(AString);
end;

function lowercase(const AString: string): string;
begin
  Result := php_strtolower(AString);
end;

procedure Write;
var
  LIndex: integer;
  AArgs: variant;
begin
  AArgs := php_func_get_args();
  for LIndex := 0 to php_count(AArgs) - 1 do begin
    php_echo(AArgs[LIndex]);
  end;
end;

procedure WriteLn;
var
  LIndex: integer;
  LArgs: variant;
begin
  LArgs := php_func_get_args();
  for LIndex := 0 to php_count(LArgs) - 1 do begin
    php_echo(LArgs[LIndex]);
  end;
  php_echo(LineEnding);
end;

initialization

end.
